### PR TITLE
[TEAM6] Rewrite treaties

### PR DESCRIPTION
**Pull request by JaafarRammal**
Proposing a cleaner treaty function code

### CODE REVIEW

**Review by jzzheng22 — APPROVED** [review]
Looks a lot cleaner, thanks

ideally put HP and AvailableFood in their own functions

**Comment by JaafarRammal** [comment]
Doing this now

File: pkg/agents/team6/treaties.go [hunk 1]
```diff
@@ -87,100 +87,82 @@ func (a *CustomAgent6) considerTreaty(t *messages.Treaty) bool {
...
+	// readable constants
+	convertedFood := a.convertToTakeFoodAmount(float64(t.ConditionValue()), t.Request(), t.RequestValue())
+	conditionCheck := t.ConditionOp() == messages.LE || t.ConditionOp() == messages.LT
+	considerTreaty := a.considerTreatyUsingUtility(t)
+
 	// We only consider meaningful LeaveAmountFood and LeavePercentFood treaties, i.e., the treaties with RequestOp GE or GT
 	if t.RequestOp() == messages.GE || t.RequestOp() == messages.GT {
 		switch t.Condition() {
 		// HP
 		case 1:
```
Comment (jzzheng22) [inline pkg/agents/team6/treaties.go:100]:
```suggestion
		case messages.HP:
```

File: pkg/agents/team6/treaties.go [hunk 2]
```diff
@@ -87,100 +87,82 @@ func (a *CustomAgent6) considerTreaty(t *messages.Treaty) bool {
...
-			case "Narcissist":
-				a.Log("I used considerTreatyUsingUtility", infra.Fields{"my floor:": a.Floor(), "my social motive": a.currBehaviour.string()})
-				return a.considerTreatyUsingUtility(t)
-			default:
-				a.Log("I used considerTreatyUsingUtility", infra.Fields{"my floor:": a.Floor(), "my social motive": a.currBehaviour.string()})
-				return a.considerTreatyUsingUtility(t)
-			}
+			return a.considerHPTreaty(conditionCheck, considerTreaty, levels, t.ConditionValue())
 		// Floor
 		case 2:
```
Comment (jzzheng22) [inline pkg/agents/team6/treaties.go:103]:
```suggestion
		case messages.Floor:
```

File: pkg/agents/team6/treaties.go [hunk 3]
```diff
@@ -87,100 +87,82 @@ func (a *CustomAgent6) considerTreaty(t *messages.Treaty) bool {
...
-			return a.considerTreatyUsingUtility(t)
+			return considerTreaty
 		// AvailableFood
 		case 3:
```
Comment (jzzheng22) [inline pkg/agents/team6/treaties.go:106]:
```suggestion
		case messages.AvailableFood:
```

**Review by jzzheng22 — CHANGES_REQUESTED** [review]

**Review by jzzheng22 — COMMENTED** [review]
much much cleaner, thanks

have added my own additional refactor as well

**Comment by jzzheng22** [comment]
@MattScottEEE  not really appropriate for me to review this given I wrote part of it, but I'm happy for this to go through. It's your team's branch though.

**Review by jzzheng22 — APPROVED** [review]

File: pkg/agents/team6/treaties.go [hunk 4]
```diff
@@ -87,87 +79,83 @@ func (a *CustomAgent6) considerTreaty(t *messages.Treaty) bool {
...
+		}
+		return true
+	case "Selfish":
+		if conditionCheck || convertedFood <= 60 {
+			return considerTreaty
+		}
+		return true
+
+	case "Narcissist":
+		return considerTreaty
```
Comment (MattScottEEE) [inline pkg/agents/team6/treaties.go:125]:
```suggestion
```

**Review by MattScottEEE — CHANGES_REQUESTED** [review]
Mat check this plz

File: pkg/agents/team6/treaties.go [hunk 5]
```diff
@@ -87,87 +79,83 @@ func (a *CustomAgent6) considerTreaty(t *messages.Treaty) bool {
...
+		}
+		return true
+
+	case "Selfish":
+		if conditionCheck || conditionValue < levels.strongLevel {
+			return considerTreaty
+		}
+		return true
+	case "Narcissist":
+		return considerTreaty
```
Comment (MattScottEEE) [inline pkg/agents/team6/treaties.go:148]:
```suggestion
```

**Review by MattScottEEE — APPROVED** [review]
Tested, LGTM - thanks Jaafar + Jason

**Pull request merged**